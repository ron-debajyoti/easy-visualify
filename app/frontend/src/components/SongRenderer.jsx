import React, { memo } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/macro';
import media from 'styled-media-query';
import PlayWidget from './Widget';

const WidgetTitle = styled.div`
  ${media.lessThan('medium')`
    font-size: small;
  `}

  ${media.greaterThan('medium')`
    font-size: large;
  `}
  margin: 30px 20px
  text-align: left;
  color: white;
`;

const WidgetWrapper = styled.div`
  display: inline;
  padding: 0.7em;
`;

const OrderedList = styled.ol`
  ${media.lessThan('medium')`
    display: inline flow-root list-item;
  `}

  ${media.greaterThan('medium')`
    display: 'inline list-item'
  `}
`;

const songObject = {
  uri: PropTypes.string.isRequired,
  popularity: PropTypes.number,
  viewCoverArt: PropTypes.bool.isRequired,
};

const propTypes = {
  content: PropTypes.arrayOf(PropTypes.arrayOf(songObject)).isRequired,
  contentType: PropTypes.string.isRequired,
};

const SongRenderer = (props) => {
  const { content, contentType } = props;
  if (content.length >= 2) {
    const populateCards1 = content[0].map((element) => (
      <PlayWidget type="song" key={element.uri} uri={element.uri} />
    ));
    const populateCards2 = content[1].map((element) => (
      <PlayWidget type="song" key={element.uri} uri={element.uri} />
    ));
    const populateCards3 = content[2].map((element) => (
      <PlayWidget type="song" key={element.uri} uri={element.uri} />
    ));
    const populateCards4 = <PlayWidget type="album" key={content[3]} uri={content[3]} />;

    if (contentType === 'top') {
      if (content[0].length > 0) {
        return (
          <WidgetWrapper>
            <WidgetTitle> Daily Top Tracks of {content[4]} </WidgetTitle>

            <OrderedList>{populateCards1}</OrderedList>
          </WidgetWrapper>
        );
      }
      return (
        <WidgetTitle>
          <WidgetTitle style={{ fontSize: '1em' }}>
            Spotify doesn&apos;t have the data yet. Try Radar Tracks
          </WidgetTitle>
        </WidgetTitle>
      );
    }
    if (contentType === 'viral') {
      if (content[1].length > 0) {
        return (
          <WidgetWrapper>
            <WidgetTitle> Daily Viral Tracks of {content[4]} </WidgetTitle>
            <OrderedList>{populateCards2}</OrderedList>
          </WidgetWrapper>
        );
      }
      return (
        <WidgetTitle>
          <WidgetTitle style={{ fontSize: '1em' }}>
            {' '}
            Spotify doesn&apos;t have the data yet. Try Radar Tracks{' '}
          </WidgetTitle>
        </WidgetTitle>
      );
    }
    if (contentType === 'weekly') {
      if (content[4].length > 0) {
        return (
          <WidgetWrapper>
            <WidgetTitle> Weekly Top Tracks of {content[4]} </WidgetTitle>
            <OrderedList>{populateCards4}</OrderedList>
          </WidgetWrapper>
        );
      }
      return (
        <WidgetTitle>
          <WidgetTitle style={{ fontSize: '1em' }}>
            {' '}
            Spotify doesn&apos;t have the data yet. Try Radar Tracks{' '}
          </WidgetTitle>
        </WidgetTitle>
      );
    }
    if (content[2].length > 0) {
      return (
        <WidgetWrapper>
          <WidgetTitle> Radar Tracks of {content[4]} </WidgetTitle>
          <div>{populateCards3}</div>
        </WidgetWrapper>
      );
    }
    return (
      <WidgetTitle>
        <WidgetTitle style={{ fontSize: '1em' }}>
          {' '}
          Spotify doesn&apos;t have the data yet. Try other Tracks{' '}
        </WidgetTitle>
      </WidgetTitle>
    );
  }

  if (content.length === 0) {
    return <WidgetTitle> Click on a country to begin! </WidgetTitle>;
  }
  return <WidgetTitle> Spotify is not supported in {content[0]} !</WidgetTitle>;
};

SongRenderer.propTypes = propTypes;

export default memo(SongRenderer);
