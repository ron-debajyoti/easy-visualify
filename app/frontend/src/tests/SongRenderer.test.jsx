import React from 'react';
import renderer from 'react-test-renderer';
import { unmountComponentAtNode } from 'react-dom';
import SongRenderer from '../components/SongRenderer';
import { fakeTrackUri, fakeTrackUri2 } from './fakedata';

const fakeListItem = {
  uri: fakeTrackUri,
  popularity: 90,
  viewCoverArt: true,
};

const fakeListItem2 = {
  uri: fakeTrackUri2,
  popularity: 50,
  viewCoverArt: false,
};

const fakeCurrentData = [
  [fakeListItem, fakeListItem2],
  [fakeListItem, fakeListItem2],
  '6sDvv5SNASdsfeeww',
  '45ba6QAtNrdv6Ke4MFOKk9',
  'United Kingdom',
];

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('renders top tracks without crashing', () => {
  const div = document.createElement('div');
  const tree = renderer
    .create(<SongRenderer content={fakeCurrentData} contentType="top" />, div)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders weekly playlist without crashing', () => {
  const div = document.createElement('div');
  const tree = renderer
    .create(<SongRenderer content={fakeCurrentData} contentType="weekly" />, div)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders corrent message at the start', () => {
  const div = document.createElement('div');
  const tree = renderer.create(<SongRenderer content={[]} contentType="" />, div).toJSON();
  expect(tree).toMatchSnapshot();
});
