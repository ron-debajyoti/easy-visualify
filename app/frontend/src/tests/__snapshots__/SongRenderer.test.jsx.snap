// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders corrent message at the start 1`] = `
.c0 {
  text-align: left;
  color: white;
}

@media (max-width:768px) {
  .c0 {
    margin: 5px 20px;
    font-size: medium;
  }
}

@media (min-width:768px) {
  .c0 {
    margin: 20px;
    font-size: large;
  }
}

<div
  className="c0"
>
   Spotify is not supported in 
   !
</div>
`;

exports[`renders top tracks without crashing 1`] = `
.c1 {
  text-align: left;
  color: white;
}

.c0 {
  display: inline;
  padding: 0.7em;
}

.c2 {
  padding: 20px;
}

@media (max-width:768px) {
  .c3 {
    min-width: 250px;
    height: 70px;
  }
}

@media (min-width:768px) {
  .c3 {
    min-width: 250px;
    height: 100px;
  }
}

@media (max-width:768px) {
  .c1 {
    margin: 5px 20px;
    font-size: medium;
  }
}

@media (min-width:768px) {
  .c1 {
    margin: 20px;
    font-size: large;
  }
}

@media (max-width:768px) {
  .c2 {
    display: -webkit-inline-box;
    display: -webkit-inline flow-root list-item;
    display: -ms-inline flow-root list-itembox;
    display: inline flow-root list-item;
  }
}

@media (min-width:768px) {
  .c2 {
    display: contents;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
     Daily Top Tracks of 
    United Kingdom
     
  </div>
  <ol
    className="c2"
  >
    <embed
      className="c3 widget-song"
      src="https://open.spotify.com/embed?uri=spotify:track:4KD0lLJ4OGonZhBeKtct9I&theme=white"
      title=""
    />
    <embed
      className="c3 widget-song"
      src="https://open.spotify.com/embed?uri=4M6XHrTc4RtdgKN9gzEAM6&theme=white"
      title=""
    />
  </ol>
</div>
`;

exports[`renders weekly playlist without crashing 1`] = `
.c2 {
  min-width: 35vh;
  min-height: 100vh;
}

.c1 {
  text-align: left;
  color: white;
}

.c0 {
  display: inline;
  padding: 0.7em;
}

@media (max-width:768px) {
  .c1 {
    margin: 5px 20px;
    font-size: medium;
  }
}

@media (min-width:768px) {
  .c1 {
    margin: 20px;
    font-size: large;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
     Weekly Top Tracks of 
    United Kingdom
     
  </div>
  <div>
    <iframe
      className="c2 widget-playlist"
      src="https://open.spotify.com/embed/playlist/45ba6QAtNrdv6Ke4MFOKk9"
      title=""
    />
  </div>
</div>
`;
